parsers/verilog.c

```diff
@@ -1824,16 +1835,19 @@ static int tagIdsInPort (tokenInfo *const token, int c, verilogKind kind, bool m
 // Tag a list of identifiers in a data declaration
 static int tagIdsInDataDecl (tokenInfo* token, int c, verilogKind kind)
 {
+	tokenInfo *module = dupToken(token); // Not a definition. Referencing a module
```

Comment:
This function is not only for a module-instantiation.
So the variable name `module` is not proper.

I am always trying to narrow a scope of variable as narrow as possible.  How about the following fix?

```diff
diff --git a/parsers/verilog.c b/parsers/verilog.c
index 73cc70da4..49ed9e52a 100644
--- a/parsers/verilog.c
+++ b/parsers/verilog.c
@@ -1835,21 +1835,19 @@ static int tagIdsInPort (tokenInfo *const token, int c, verilogKind kind, bool m
 // Tag a list of identifiers in a data declaration
 static int tagIdsInDataDecl (tokenInfo* token, int c, verilogKind kind)
 {
-       tokenInfo *module = dupToken(token); // Not a definition. Referencing a module
-       module->kind = K_MODULE;                         // for defining an instance.
-
        c = skipClassType (token, c);
        if (c == ';')
-               goto out;
+               return c;
 
        // skip drive|charge strength or type_reference, dimensions, and delay for net
        if (c == '(')
                c = skipPastMatch ("()");
        c = skipDimension (c);
        if (c == '.')
-               goto out;       // foo[...].bar = ..;
+               return c;       // foo[...].bar = ..;
        c = skipDelay (token, c);       // ## (cycle delay)
 
+       tokenInfo *token_saved = dupToken(token); // maybe a module_identifier
        while (c != EOF)
        {
                bool with = false;
@@ -1863,7 +1861,7 @@ static int tagIdsInDataDecl (tokenInfo* token, int c, verilogKind kind)
                        if (c == '=')
                                c = skipExpression (c);
                }
-               else if (c == '(' || c == '[')  // should be instance
+               else if (c == '(' || c == '[')  // should be an instance
                {
                        c = skipDimension (c); // name_of_instance {unpacked_dimension}
                        c = skipPastMatch ("()"); // list_of_port_connections
@@ -1872,8 +1870,9 @@ static int tagIdsInDataDecl (tokenInfo* token, int c, verilogKind kind)
                        // var `add_t(foo) = '0;
                        if (c == ';' || c == ',')
                        {
+                               token_saved->kind = K_MODULE;   // for type_ref field
                                verbose ("find instance: %s with kind %s\n", vStringValue (token->name), getNameForKind (K_I
NSTANCE));
-                               createTagFull (token, K_INSTANCE, module);
+                               createTagFull (token, K_INSTANCE, token_saved);
                        }
                }
                c = skipMacro (c, token);       // `ifdef, `else, `endif, etc. (before comma)
@@ -1882,8 +1881,7 @@ static int tagIdsInDataDecl (tokenInfo* token, int c, verilogKind kind)
                c = skipWhite (vGetc ());       // skip ','
                c = skipMacro (c, token);       // `ifdef, `else, `endif, etc. (after comma)
        }
- out:
-       deleteToken (module);
+       deleteToken (token_saved);
        return c;
 }
```